lib/hexdocs/queue.ex

```diff
@@ -250,4 +266,96 @@ defmodule Hexdocs.Queue do
       )
     end)
   end
+
+  defp find_search_items(package, version, files) do
+    search_data =
+      Enum.find_value(files, fn {path, content} ->
+        case Path.basename(path) do
+          "search_data-" <> _digest -> {:elixir, content}
+          "search-data.js" -> {:gleam, content}
+          _other -> nil
+        end
+      end)
+
+    case search_data do
+      {:elixir, content} ->
+        extract_elixir_search_items(package, version, content)
+
+      {:gleam, content} ->
+        extract_gleam_search_items(package, version, content)
+
+      nil ->
+        Logger.info("Failed to find search data for #{package} #{version}")
+        nil
+    end
+  end
+
+  defp extract_gleam_search_items(package, version, search_data) do
+    "window.Gleam.initSearch(" <> search_items_js = search_data
+
+    search_items =
+      case Jason.decode(search_items_js) do
```

Comment:
Should we strip the trailing parens before trying to decode? Or is it not so trivial?

Reply:
> Should we strip the trailing parens before trying to decode?

Yes, that's how it should've been done :)